src/quafel/pipelines/data_science/frameworks.py

```diff
@@ -349,3 +259,248 @@ def get_result(self) -> Dict[str, float]:
                 counts[bitstring] = 0
 
         return counts
+
+
+class qulacs_fw:
+    def __init__(self, qasm_circuit, n_shots):
+        self.n_qubits = calculate_n_qubits_from_qasm(qasm_circuit)
+        # removing measurment operation...
+        qasm_circuit = re.sub("\\nmeasure .*;", "", qasm_circuit)
+        # ... and creg operation as they are not supported by qulacs
+        # see `convert_QASM_to_qulacs_circuit` for details.
+        qasm_circuit = re.sub("\\ncreg .*;", "", qasm_circuit)
```

Comment:
Without having looked at the qasm_circuit generation myself, this could result in unexpected behavior. I suspect that a qasm_circuit is generated based on a seed. A user would expect that two frameworks, using the same seed for the circuit, will be evaluated on the **exact** same circuit. 

Could this be a problem? You could just throw a simple warning, that the circuit has to be changed. If there are similar restrictions for other frameworks, it might be beneficial to change the circuit generation.

(If the circuit generation is way more fancy or does not use a seed anymore, you can of course ignore this comment :smile: )

Reply:
good point! However, this should not be an issue, because
- measure and creg tags do not change the overall unitary of the circuit
- always all qubits are being measured (so no resulting mixed states)
- in this case, I remove the mentioned strings from the qasm circuit, but add the same functionality later manually, just because Qulacs cannot handle those particular strings (again no functionality changed compared to other fws)
- circuit generation is still the same for all fws, because seeding is done prior to this conversion